test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/contract/HapiContractCall.java

```diff
@@ -245,14 +261,8 @@ protected Consumer<TransactionBody.Builder> opBodyDef(HapiApiSpec spec) throws T
 		if (explicitHexedParams.isPresent()) {
 			callData = explicitHexedParams.map(Supplier::get).map(CommonUtils::unhex).get();
 		} else {
-			final var paramsList = Arrays.asList(params);
-			final var tupleExist = paramsList.stream().anyMatch(p -> p instanceof Tuple || p instanceof Tuple[]);
-			if (tupleExist) {
-				callData = encodeParametersWithTuple(params);
-			} else {
-				callData = (!abi.equals(FALLBACK_ABI))
-						? CallTransaction.Function.fromJsonInterface(abi).encode(params) : new byte[] { };
-			}
+			callData = (!abi.equals(FALLBACK_ABI))
+					? encodeParametersWithTuple(abi, params) : new byte[] { };
```

Comment:
Now that we have a single place to encode parameters, I think we can rename the function `encodeParametersWithTuple` -> `encodeParameters`